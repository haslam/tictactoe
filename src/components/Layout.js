import React from 'react'
import styles from '../styles/layout.module.css'
import Board from './Board'

export default function Layout () {
  return (
    <div className={styles.layout}>
      <div className={styles.layoutBox}>
        <div>
          <h1 className={styles.appTitle}>TIC TAC TOE</h1>
          <Board />
        </div>
      </div>
    </div>
  )
}
