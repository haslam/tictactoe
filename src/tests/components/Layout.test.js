import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../../components/Layout';

describe('Layout', () => {
  const wrapper = shallow(<Layout />);
  it('should have TIC TAC TOE title', () => {
    expect(wrapper.find('h1').text()).toContain('TIC TAC TOE');
  });

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  })
})