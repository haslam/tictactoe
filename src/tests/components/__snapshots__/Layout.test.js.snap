// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="layoutBox"
        >
          <div>
            <h1
              className="appTitle"
            >
              TIC TAC TOE
            </h1>
            <Board />
          </div>
        </div>,
        <nav />,
      ],
      "className": "layout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <h1
              className="appTitle"
            >
              TIC TAC TOE
            </h1>
            <Board />
          </div>,
          "className": "layoutBox",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="appTitle"
              >
                TIC TAC TOE
              </h1>,
              <Board />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "TIC TAC TOE",
                "className": "appTitle",
              },
              "ref": null,
              "rendered": "TIC TAC TOE",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "nav",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="layoutBox"
          >
            <div>
              <h1
                className="appTitle"
              >
                TIC TAC TOE
              </h1>
              <Board />
            </div>
          </div>,
          <nav />,
        ],
        "className": "layout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <h1
                className="appTitle"
              >
                TIC TAC TOE
              </h1>
              <Board />
            </div>,
            "className": "layoutBox",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="appTitle"
                >
                  TIC TAC TOE
                </h1>,
                <Board />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "TIC TAC TOE",
                  "className": "appTitle",
                },
                "ref": null,
                "rendered": "TIC TAC TOE",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "nav",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
